### ISSUE TITLE
Cannot compile in release mode.

### ISSUE THREAD

**Issue by romanvolkov-io** [dekatotoro/SlideMenuControllerSwift#273]
If compiled in release mode such as App Store or Ad Hoc deployments, SlideMenuControllerSwift doesn't work because the linker doesn't find symbols, etc.

Extensions such as these cause syntax errors like “ambiguous use” if the Swift file is being directly copied to the project.
@interface UIViewController (SWIFT_EXTENSION(SlideMenuControllerSwift))
- (SlideMenuController * _Nullable)slideMenuController;
- (void)addLeftBarButtonWithImage:(UIImage * _Nonnull)buttonImage;
- (void)addRightBarButtonWithImage:(UIImage * _Nonnull)buttonImage;
- (void)toggleLeft;
- (void)toggleRight;
- (void)openLeft;
- (void)openRight;
- (void)closeLeft;
- (void)closeRight;
- (void)addPriorityToMenuGesuture:(UIScrollView * _Nonnull)targetScrollView;
@end

Curiously, everything works just fine within the demo project; and project settings seem to be identical.

Any ideas? Maybe compiler optimized things away?

**Comment by timlehr** [comment]
I have the same issue with Xcode 8 and Swift 3.0. Installed SMCS via Carthage. Have you solved the issue yet? Nothing seems to work for me. :(

**Comment by romanvolkov-io** [comment]
No, compiler still doesn't like the code. Go with https://github.com/Friend-LGA/LGSideMenuController instead.

**Comment by timlehr** [comment]
Your comment gave me some inspiration that it might be an issue with the optimization and as soon as I turned off Whole Module Optimization for my project and set it to `None -Onone` in the build settings, the archive is built. Sucks, but I guess it’s better than nothing for now.

To the author: Please check this issue. :)

**Comment by obuseme** [comment]
Turning off Whole Module Optimization worked for my project as well, though not ideal. What's the "real" fix for this?

**Comment by Nomad00** [comment]
Turning off Whole Module Optimization for the Framework target worked in my fork (https://github.com/Nomad00/SlideMenuControllerSwift/tree/no-optimization). Definitely not the solution, but a workaround for now.

**Comment by biagionicp** [comment]
same thing happens to be but when compiling in debug mode
